api4/team.go

```diff
@@ -967,16 +969,26 @@ func inviteUsersToTeam(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	err := c.App.InviteNewUsersToTeam(emailList, c.Params.TeamId, c.App.Session.UserId)
-	if err != nil {
-		c.Err = err
-		return
+	if graceful {
+		invitesWithError, err := c.App.InviteNewUsersToTeamGracefully(emailList, c.Params.TeamId, c.App.Session.UserId)
+		if err != nil {
+			c.Err = err
+			return
+		}
+		// in graceful mode we return both the succesful ones and the failed ones
+		w.Write([]byte(model.EmailInviteWithErrorToJson(invitesWithError)))
+	} else {
```

Comment:
My question is in what scenario would we want a non-graceful send? Why couldn't we just always do a graceful invite?

Reply:
API compatibility